rfcs/0000-expressive-diagnostics.md

```diff
@@ -0,0 +1,356 @@
+- Feature Name: `expressive_diagnostics`
+- Start Date: 2023-08-07
+- RFC PR: [FuelLabs/sway-rfcs#0000](https://github.com/FuelLabs/sway-rfcs/pull/001)
```

Comment:
```suggestion
- RFC PR: [FuelLabs/sway-rfcs#0010](https://github.com/FuelLabs/sway-rfcs/pull/030)
```

Reply:
Thanks for the suggested change! I hope this is exactly how it will look like at the end :-) but let's stick to the process and see if the RFC will get accepted.